vignettes/meetupr.Rmd

```diff
@@ -0,0 +1,223 @@
+```
+
+This function will show you:  
+- Which authentication method is active  
+- Whether your credentials are properly configured  
+- API connectivity status  
+- Setup instructions if authentication is missing  
+
+## Authentication Methods
+
+meetupr supports two authentication methods:
+
+### OAuth Authentication (Interactive)
+
+Best for interactive development, data exploration, and one-off analyses.
+
+**Setup:**
+```r
+# Authenticate interactively (opens browser)
+meetup_auth()
+
+# Check authentication status
+meetup_auth_status()
+```
+
+The OAuth token is cached locally in your httr2 cache directory. 
+Token files follow the pattern: `{hash}-token.rds.enc`
+
+### CI/CD Authentication (Non-Interactive)
+
+For automated workflows, you can set up token-based authentication:
+
+**1. Authenticate locally first:**
+```r
+meetup_auth()
+meetup_auth_setup_ci()
+```
+
+This encodes your token as base64 and provides instructions for setting CI secrets.
+
+**2. Set environment variables in your CI:**
+- `MEETUP_TOKEN` - Base64-encoded token (from step 1)
+- `MEETUP_TOKEN_FILE` - Token filename (e.g., `ae743e0-token.rds.enc`)
+
+**3. Token loads automatically in CI** 
+
+To make the authentication work in CI/CD pipelines, add this line before running any API calls:
+
+```r
+meetup_auth_load_ci()
+```
+
+
+
+## Environment Variables Reference
+
+| Variable | Required | Purpose |
+|----------|----------|---------|
+| `MEETUP_CLIENT_ID` | Optional* | OAuth client ID |
+| `MEETUP_CLIENT_SECRET` | Optional* | OAuth client secret |
+| `MEETUP_CLIENT_NAME` | No | OAuth client name (default: "meetupr") |
+| `MEETUP_TOKEN` | for CI | Base64-encoded OAuth token |
+| `MEETUP_TOKEN_FILE` | for CI | Token filename |
+| `MEETUPR_DEBUG` | for CI | Enable debug output ("1" = on) |
+
+\* Built-in credentials are used if not provided, and Meetup clients are only available for approved apps from Pro (paid) accounts.
+
+### Setting Environment Variables
+
+If you want to use your own Meetup OAuth app credentials, set them as environment variables.
+Use `.Renviron` for local development:
+
+```r
+usethis::edit_r_environ()
+```
+
+Add:
+
+```bash
+MEETUP_CLIENT_ID=your_client_id
+MEETUP_CLIENT_SECRET=your_secret
+```
+
+The `.Renviron` file contains potentially sensitive credentials.
+While the client ID and secret are not as sensitive as the token, treat them with care.
+However, despite secrets generally should remain secret, noone can actually use the Meetup API without authenticating as a user, so the risk is lower. 
+
+### Verification
+
+After setting environment variables, restart R and check:
+
+```{r}
+#| label: verify-sitrep
+#| cassette: true
+meetup_sitrep()
+```
+
+## Basic Usage Examples
+
+### Get Group Information
+
+```{r}
+#| label: group-info
+#| cassette: true
+group_info <- get_group("rladies-lagos")
+str(group_info)
+```
+
+### List Upcoming Events
+
+```{r}
+#| label: upcoming-events
+#| cassette: true
+events <- get_events("rladies-lagos")
```

Comment:
I tried `get_events("rladies-ottawa")` because I had just read about https://www.meetup.com/rladies-ottawa/events/311256371/?eventOrigin=group_upcoming_events, but I only got 6 past events.